src/abilities/publisherAbility.ts

```diff
@@ -1,21 +1,21 @@
 import { defineAbility } from '@casl/ability'
 
-export default defineAbility((can) => {
+export default defineAbility((can, cannot) => {
   // Market
   can('browse', 'market')
-  can('consume', 'market')
-  can('compute', 'market')
   can('access', 'market')
   can('update', 'market')
   can('publish', 'market')
   can('delete', 'market')
+  cannot('consume', 'market')
```

Comment:
Why don't you want a publisher to be able to consume?

Reply:
@jamiehewitt15  because both consume and publish are distinct capability.

In some scenario, the user only will have publisher role, thus, they are just able to publish, this provide better control on the capability.  

However, the same user can have publisher and consumer role.
If this case, this user could do both.

Reply:
Ok sure, sounds fine to me